### PR TITLE
FEAT(FTDA): Create a login page that uses basic string validation to log in user

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**Description of Changes**

- Created Login page 
- Home Page Get started button now routes to login page 
- Added route in Login page to main application if validation succeeds
- Used preset values for email and password for basic string validation when logging in 

https://github.com/user-attachments/assets/70b4116d-ba01-4df8-92b1-1b261bbc0a48

### CODE REVIEW

File: pages/login.page.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import LoginPageContainer from '@/modules/LoginPage/containers/LoginPageContainer'
+import React from 'react'
+
+const LoginPage: React.FC = () => {
+  return (
+    <LoginPageContainer/>
+
```
Comment (ragib-hossain-sazim) [inline pages/login.page.tsx]: [Nitpick] Unnecessary new line

File: pages/login.page.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import React from 'react'
+
+const LoginPage: React.FC = () => {
+  return (
+    <LoginPageContainer/>
+   
+  )
+}
+
+export default LoginPage
```
Comment (ragib-hossain-sazim) [inline pages/login.page.tsx]: Please Add semicolons after end of lines.

**Comment by ragib-hossain-sazim** [comment]
Usually we don't specify what part of the credential is wrong to the user. This is considered a security issue. You can just show the invalid credentials error maybe at the top of the card.

File: modules/LoginPage/containers/LoginPageContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        if (value !== predefinedEmail) {
+          return "Invalid Email";
+        }
+      },
+      password: (value) => {
+        if (!value) {
+          return "Password is required";
+        }
+        if (value !== predefinedPassword) {
```
Comment (ragib-hossain-sazim) [inline modules/LoginPage/containers/LoginPageContainer.tsx]: this rule won't be needed here

File: modules/LoginPage/containers/LoginPageContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    initialValues: {
+      email: "",
+      password: "",
+    },
+    validate: {
+      email: (value) => {
+        if (!value) {
+          return "Email is required";
+        }
+        if (value !== predefinedEmail) {
```
Comment (ragib-hossain-sazim) [inline modules/LoginPage/containers/LoginPageContainer.tsx]: this rule won't be needed here.

File: modules/LoginPage/containers/LoginPageContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+import classes from "./LoginPage.module.css";
+
+const predefinedEmail = "test@gmail.com";
+const predefinedPassword = "123456";
+
+const validateLogin = (email: string, password: string, router: NextRouter) => {
+  if (email === predefinedEmail && password === predefinedPassword) {
+    router.push("/todo");
+  }
```
Comment (ragib-hossain-sazim) [inline modules/LoginPage/containers/LoginPageContainer.tsx]: You can handle the credential mismatch issue and trigger the error from here.

**Comment by yamin-faruque-sazim** [comment]
Addressed PR comments and pushed in new changes

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Suggestion: You can look into configuring prettier and eslint to ensure the formatting of imports, missing semicolons etc. Best to depend on tooling for these things which are easy to miss.
Can find those configured in the sazim template repos.

**Pull request merged**